test/micromamba/test_shell.py

```diff
@@ -145,68 +157,65 @@ def custom_shell(shell):
     @pytest.mark.parametrize("expanded_home", [False, True])
     @pytest.mark.parametrize("prefix_type", ["prefix", "name"])
     def test_activate(self, shell_type, root, env_exists, prefix_type, expanded_home):
-        if (
-            (platform.system() == "Linux" and shell_type not in ("bash", "posix"))
-            or (
-                platform.system() == "Windows"
-                and shell_type not in ("cmd.exe", "powershell")
-            )
-            or (
-                platform.system() == "Darwin"
-                and shell_type not in ("zsh", "bash", "posix")
-            )
-        ):
-            pytest.skip("Incompatible shell/OS")
+        skip_if_shell_incompat(shell_type)
 
-        if not root and env_exists:
+        if env_exists:
+            # Create the environment for this test, so that it exists
             create("-n", TestShell.env_name, "-q", "--offline", no_dry_run=True)
+        else:
+            shutil.rmtree(TestShell.root_prefix)
```

Comment:
Shouldn't this be handled by the `teardown`?

Reply:
there are many ways to do, but since it's the only test that needs to sometimes a missing root_prefix, I prefer that way

Reply:
I think this is unnecessarily complex as the starting state and end state of the test are ambiguous.